static/src/javascripts/projects/common/modules/commercial/contributions-service.js

```diff
@@ -0,0 +1,210 @@
+};
+
+interface InitAutomatJsConfig {
+    epicRoot: HTMLElement | ShadowRoot;
+    onReminderOpen?: Function;
+}
+
+interface AutomatJsCallback {
+    buttonCopyAsString: string;
+}
+
+// TODO introduce better way to support client-side behaviour
+const executeJS = (container: HTMLElement | ShadowRoot, js: string) => {
+    if (!js) {
+        return;
+    }
+
+    try {
+        // eslint-disable-next-line no-eval
+        window.eval(js);
+        if (
+            typeof window.initAutomatJs ===
+            'function'
+        ) {
+            const initAutomatJsConfig: InitAutomatJsConfig = {
+                epicRoot: container,
+                onReminderOpen: (callbackParams: AutomatJsCallback) => {
+                    const { buttonCopyAsString } = callbackParams;
+                    submitClickEvent({
+                        component: {
+                            componentType: 'ACQUISITIONS_OTHER',
+                            id: 'precontribution-reminder-prompt-clicked',
+                        },
+                    });
+                    submitClickEvent({
+                        component: {
+                            componentType: 'ACQUISITIONS_OTHER',
+                            id: `precontribution-reminder-prompt-copy-${buttonCopyAsString}`,
+                        },
+                    });
+                },
+            };
+            window.initAutomatJs(initAutomatJsConfig);
+        }
+    } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error(error);
+        reportError(error, {}, false);
+    }
+};
+
+const buildPayload = () => {
+    const ophan = config.get('ophan');
+    const page = config.get('page');
+
+    // note, there is a race condition so we want to fetch this as late as possible to give a change for the geo local storage value to be set
+    const countryCode = geolocationGetSync();
+
+    const tracking = {
+        ophanPageId: ophan.pageViewId,
+        ophanComponentId: 'ACQUISITIONS_EPIC',
+        platformId: 'GUARDIAN_WEB',
+        clientName: 'frontend',
+        referrerUrl:
+            window.location.origin + window.location.pathname,
+    };
+
+    const targeting = {
+        contentType: page.contentType,
+        sectionName: page.section,
+        shouldHideReaderRevenue: page.shouldHideReaderRevenue,
+        isMinuteArticle: config.hasTone('Minute'),
+        isPaidContent: page.isPaidContent,
+        isSensitive: page.isSensitive,
+        tags: buildKeywordTags(page),
+        // This test is already subjected to the 3 checks below, but
+        // we're passing these properties to the Contributions
+        // service for consistency with DCR.
+        showSupportMessaging: !shouldNotBeShownSupportMessaging(),
+        isRecurringContributor: isRecurringContributor(),
+        lastOneOffContributionDate:
+            getLastOneOffContributionDate() || undefined,
+        mvtId: getMvtValue(),
+        countryCode,
+    };
```

Comment:
I think there's two properties missing here in the payload (the helper functions are exposed via client lib):
`epicViewLog: getViewLog(),`
`weeklyArticleHistory: getWeeklyArticleHistory(),`

These are already in the comparison payload and also in the Epic payload in DCR, but were missing in the `FrontendDotcomRenderingEpic` test, which makes me very confused. How was our service applying those max viewing rules (4 epics in 30 days) without an Epic view log? 🤔

Reply:
Yeah, to be honest I just copied this code from FrontendDotcomRenderingEpic. Will fix.